src/screens/DeviceStack/KeystoneStack/keystone-accounts.tsx

```diff
@@ -0,0 +1,180 @@
+import React, {memo, useCallback, useMemo, useRef, useState} from 'react';
+
+import {ProviderKeystoneReactNative} from '@haqq/provider-keystone-react-native';
+
+import {KeystoneAccounts} from '@app/components/keystone/keystone-accounts';
+import {app} from '@app/contexts';
+import {showModal} from '@app/helpers';
+import {AddressUtils} from '@app/helpers/address-utils';
+import {getWalletsFromProvider} from '@app/helpers/get-wallets-from-provider';
+import {useTypedNavigation, useTypedRoute} from '@app/hooks';
+import {useEffectAsync} from '@app/hooks/use-effect-async';
+import {Wallet} from '@app/models/wallet';
+import {navigator} from '@app/navigator';
+import {
+  KeystoneStackParamList,
+  KeystoneStackRoutes,
+  SignInStackParamList,
+} from '@app/route-types';
+import {Balance} from '@app/services/balance';
+import {Indexer} from '@app/services/indexer';
+import {ChooseAccountItem, ModalType, WalletType} from '@app/types';
+
+const PAGE_SIZE = 5;
+
+export const KeystoneAccountsScreen = memo(() => {
+  const navigation = useTypedNavigation<SignInStackParamList>();
+  const {qrCBORHex, ...params} = useTypedRoute<
+    KeystoneStackParamList,
+    KeystoneStackRoutes.KeystoneAccounts
+  >().params;
+  const provider = useRef(
```

Comment:
```suggestion
  const provider = getProviderInstanceForWallet
```

Reply:
it's not possible, because I don't have wallet instance here

```ts
const provider = useRef(
    new ProviderKeystoneReactNative({
      qrCBORHex,
      awaitForSign: Promise.resolve,
    }),
  );
```